geode-core/src/distributedTest/java/org/apache/geode/internal/cache/PartitionedRegionRestartRebalanceDUnitTest.java

```diff
@@ -0,0 +1,133 @@
+
+public class PartitionedRegionRestartRebalanceDUnitTest implements Serializable {
+  private static final int REDUNDANT_COPIES = 2;
+  private static final int TOTAL_NUM_BUCKETS = 12;
+  private static final Logger logger = LogManager.getLogger();
+
+  private String REGION_NAME = getClass().getSimpleName();;
+  private VM[] datastores;
+
+  @Rule
+  public DistributedRule distributedRule = new DistributedRule();
+
+  @Rule
+  public CacheRule cacheRule = new CacheRule();
+
+  @Before
+  public void setUp() throws Exception {
+    datastores = new VM[4];
+    for (int i = 0; i < datastores.length; i++) {
+      datastores[i] = getVM(i);
+      datastores[i].invoke(() -> cacheRule.createCache());
+      datastores[i].invoke(() -> createRegion());
+    }
+    datastores[0].invoke(() -> feedData());
+  }
+
+  private void createRegion() {
+    PartitionAttributesFactory<String, Integer> paf = new PartitionAttributesFactory();
+    paf.setRedundantCopies(REDUNDANT_COPIES);
+    paf.setTotalNumBuckets(TOTAL_NUM_BUCKETS);
+
+    RegionFactory<String, Integer> rf = cacheRule.getCache().createRegionFactory();
+    rf.setDataPolicy(DataPolicy.PARTITION);
+    rf.setPartitionAttributes(paf.create());
+    LocalRegion region = (LocalRegion) rf.create(REGION_NAME);
+  }
+
+  private void feedData() throws InterruptedException {
+    PartitionedRegion pr = (PartitionedRegion) cacheRule.getCache().getRegion(REGION_NAME);
+    for (int i = 0; i < TOTAL_NUM_BUCKETS * 2; i++) {
+      pr.put(i, "VALUE-" + i);
+      if (i < TOTAL_NUM_BUCKETS) {
+        pr.destroy(i);
+      }
+    }
+    cacheRule.getCache().getTombstoneService().forceBatchExpirationForTests(TOTAL_NUM_BUCKETS);
+  }
+
+  private void rebalance() throws InterruptedException {
+    RebalanceOperation op =
+        cacheRule.getCache().getResourceManager().createRebalanceFactory().start();
+    RebalanceResults results = op.getResults();
+    logger.info("Rebalance total time is " + results.getTotalTime());
+  }
+
+  private void verify() {
+    PartitionedRegion pr = (PartitionedRegion) cacheRule.getCache().getRegion(REGION_NAME);
+    for (BucketRegion br : pr.getDataStore().getAllLocalBucketRegions()) {
+      Set<VersionSource> departedMemberSet = br.getVersionVector().getDepartedMembersSet();
+      for (Object key : br.getRegionKeysForIteration()) {
+        RegionEntry entry = br.getRegionEntry(key);
+        departedMemberSet.remove(entry.getVersionStamp().getMemberID());
+        if (departedMemberSet.isEmpty()) {
+          break;
+        }
+      }
+      Map map = br.getVersionVector().getMemberToVersion();
+      for (Object key : br.getVersionVector().getMemberToVersion().keySet()) {
+        logger.info(br.getFullPath() + ":" + key + ":"
+            + br.getVersionVector().getMemberToVersion().get(key));
+      }
+      // The test proved that departedMemberSet is not growing
+      assertThat(departedMemberSet.size()).isLessThanOrEqualTo(datastores.length);
```

Comment:
`departedMemberSet` is a set, which does not allow duplicate element. The maximal number of departed member is `datastores`' length. Therefore `departedMemberSet` is always less than or equal to `datastores.length`. Better find other assertion I think.